tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/AwsKinesisCommon.cs

```diff
@@ -22,6 +22,22 @@ internal static class AwsKinesisCommon
         internal const string IntegrationName = nameof(IntegrationId.AwsKinesis);
         internal const IntegrationId IntegrationId = Configuration.IntegrationId.AwsKinesis;
 
+        public static string? StreamNameFromARN(string? arn)
+        {
+            if (string.IsNullOrEmpty(arn))
+            {
+                return null;
+            }
+
+            var arnComponents = arn!.Split('/');
```

Comment:
[performance] Not necessarily a blocker (we could come back and clean it up in a separate PR later), but if this code is in the app's hot path, we should do this in a way that doesn't allocate the `arnComponents` string array.

One approach could be to use `IndexOf('/')` twice. We also have a `string.SplitIntoSpans()` helper that splits a string into `Span<char>` tokens and enumerate over them.

Here's an example that uses both methods: https://github.com/DataDog/dd-trace-dotnet/blob/eed7b60720fd1df2a34e218761b17a13d5c6a14f/tracer/src/Datadog.Trace/Propagators/W3CBaggagePropagator.cs#L250-L256

Reply:
I've added this to DSMON-490 tracking followup improvements, planned to be a fast-follow after the freeze.